common/views.py

```diff
@@ -57,14 +55,10 @@ def dispatch(self, request, *args, **kwargs):
         if not self.application:
             return self.error_response(self.application_error)
 
-        querydict = querydict_from_dict({
-            'next': request.get_full_path(),
-            'connection': str(self.application.name)
-        })
-
-        url = reverse('account-login-oauth2')
-
-        url_parts = list(urllib.parse.urlparse(url))
-        url_parts[4] = querydict.urlencode()
+        # Assume that we are coming in from an oauth2 project and go ahead and
+        # clobber the redirect url in the session.
+        next_t = '{0}&connection={1}'.format(request.get_full_path(),
```

Comment:
I don't think you mean to store connection in session as well. All you want to store is the URL we want to redirect to and skip trying to manage "next" parameters around login/signup – that was the simplification we seek. :)

So the querydict above can the `next` parameter, but retain the `connection` parameter.